lib/Sabre/PublicAlbumAuthBackend.php

```diff
@@ -21,46 +21,58 @@
  */
 namespace OCA\Photos\Sabre;
 
-use OC\Security\Bruteforce\Throttler;
-use OCA\Photos\Album\AlbumMapper;
-use OCP\IRequest;
-use Sabre\DAV\Auth\Backend\AbstractBasic;
+use Sabre\DAV\Auth\Backend\BackendInterface;
+use Sabre\HTTP\RequestInterface;
+use Sabre\HTTP\ResponseInterface;
 
-class PublicAlbumAuthBackend extends AbstractBasic {
-	private const BRUTEFORCE_ACTION = 'publicphotos_webdav_auth';
-	private IRequest $request;
-	private AlbumMapper $albumMapper;
-	private Throttler $throttler;
-
-	public function __construct(
-		IRequest $request,
-		AlbumMapper $albumMapper,
-		Throttler $throttler
-	) {
-		$this->request = $request;
-		$this->albumMapper = $albumMapper;
-		$this->throttler = $throttler;
+class PublicAlbumAuthBackend implements BackendInterface {
+	public function __construct() {
 	}
 
 	/**
-	 * Validates the token.
+	 * When this method is called, the backend must check if authentication was
+	 * successful.
+	 *
+	 * The returned value must be one of the following
+	 *
+	 * [true, "principals/username"]
+	 * [false, "reason for failure"]
+	 *
+	 * If authentication was successful, it's expected that the authentication
+	 * backend returns a so-called principal url.
+	 *
+	 * Examples of a principal url:
+	 *
+	 * principals/admin
+	 * principals/user1
+	 * principals/users/joe
+	 * principals/uid/123457
+	 *
+	 * If you don't use WebDAV ACL (RFC3744) we recommend that you simply
+	 * return a string such as:
+	 *
+	 * principals/users/[username]
 	 *
-	 * @param string $username
-	 * @return bool
-	 * @throws \Sabre\DAV\Exception\NotAuthenticated
+	 * @return array
 	 */
-	protected function validateUserPass($username, $password) {
-		$this->throttler->sleepDelayOrThrowOnMax($this->request->getRemoteAddress(), self::BRUTEFORCE_ACTION);
-
-		$albums = $this->albumMapper->getSharedAlbumsForCollaboratorWithFiles($username, AlbumMapper::TYPE_LINK);
-
-		if (count($albums) !== 1) {
-			$this->throttler->registerAttempt(self::BRUTEFORCE_ACTION, $this->request->getRemoteAddress());
-			return false;
-		}
-
-		\OC_User::setIncognitoMode(true);
+	public function check(RequestInterface $request, ResponseInterface $response) {
+		\OC_User::setIncognitoMode(true); // ???
+		return [true, "principals/token"];
```

Comment:
I understand that we need to replace 'token' with the actual token in the returned value.

```suggestion
	public function check(RequestInterface $request, ResponseInterface $response) {
		\OC_User::setIncognitoMode(true);

		$auth = new Sabre\HTTP\Auth\Basic(
			$this->realm,
			$request,
			$response
		);

		$userpass = $auth->getCredentials();
		if (!$userpass) {
			return [false, "No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured"];
		}

		return [true, "principals/".$userpass[0]];
	}
```

Reply:
I am not sure what is the meaning of the return value in this case. I have copied the comment above the `check` method from the original `Sabre\DAV\Auth\Backend\BackendInterface` file: https://github.com/sabre-io/dav/blob/master/lib/DAV/Auth/Backend/BackendInterface.php#L19
I suspect that the returned value is not used at all. Only when the `check` method returns `[false, "error msg"]` the request is rejected.

Actual token verification is performed in `PublicRootCollection`:
https://github.com/nextcloud/photos/blob/4c87d2566bdf405bb5f430cc8fd839be64587064/lib/Sabre/PublicRootCollection.php#L92

Regarding bringing back the `Sabre\HTTP\Auth\Basic` class and the `getCredentials` call - this is exactly what I wanted to avoid when suggesting changes to the `PublicAlbumAuthBackend`. We **want the request to succeed** for anonymous users - i.e. when there is no `Authorization: Basic` header. This was the reason why the original code failed when an anonymous user tried to download pictures from a shared album.

Reply:
You're right, let's keep your version :)